File: goclient/item/provider_item_request_builder.go
package item

import (
    "context"
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f "github.com/microsoft/kiota-abstractions-go"
    iae9fb0262d34e3cd98c53508ace559228b424528fd79025950263229053f76e2 "github.com/andrueastman/untypednodesamples/goclient/models"
)

// ProviderItemRequestBuilder builds and executes requests for operations under \{provider-id}
type ProviderItemRequestBuilder struct {
    i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.BaseRequestBuilder
}
// ProviderItemRequestBuilderGetRequestConfiguration configuration for the request such as headers, query parameters, and middleware options.
type ProviderItemRequestBuilderGetRequestConfiguration struct {
    // Request headers
    Headers *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestHeaders
    // Request options
    Options []i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestOption
}
// NewProviderItemRequestBuilderInternal instantiates a new ProviderItemRequestBuilder and sets the default values.
func NewProviderItemRequestBuilderInternal(pathParameters map[string]string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ProviderItemRequestBuilder) {
    m := &ProviderItemRequestBuilder{
        BaseRequestBuilder: *i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewBaseRequestBuilder(requestAdapter, "{+baseurl}/{provider%2Did}", pathParameters),
    }
    return m
}
// NewProviderItemRequestBuilder instantiates a new ProviderItemRequestBuilder and sets the default values.
func NewProviderItemRequestBuilder(rawUrl string, requestAdapter i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestAdapter)(*ProviderItemRequestBuilder) {
    urlParams := make(map[string]string)
    urlParams["request-raw-url"] = rawUrl
    return NewProviderItemRequestBuilderInternal(urlParams, requestAdapter)
}
// Get list all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
// returns a APIsable when successful
func (m *ProviderItemRequestBuilder) Get(ctx context.Context, requestConfiguration *ProviderItemRequestBuilderGetRequestConfiguration)(iae9fb0262d34e3cd98c53508ace559228b424528fd79025950263229053f76e2.APIsable, error) {
    requestInfo, err := m.ToGetRequestInformation(ctx, requestConfiguration);
    if err != nil {
        return nil, err
    }
    res, err := m.BaseRequestBuilder.RequestAdapter.Send(ctx, requestInfo, iae9fb0262d34e3cd98c53508ace559228b424528fd79025950263229053f76e2.CreateAPIsFromDiscriminatorValue, nil)
    if err != nil {
        return nil, err
    }
    if res == nil {
        return nil, nil
    }
    return res.(iae9fb0262d34e3cd98c53508ace559228b424528fd79025950263229053f76e2.APIsable), nil
}
// ServicesJson the servicesJson property
// returns a *ServicesJsonRequestBuilder when successful
func (m *ProviderItemRequestBuilder) ServicesJson()(*ServicesJsonRequestBuilder) {
    return NewServicesJsonRequestBuilderInternal(m.BaseRequestBuilder.PathParameters, m.BaseRequestBuilder.RequestAdapter)
}
// ToGetRequestInformation list all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
// returns a *RequestInformation when successful
func (m *ProviderItemRequestBuilder) ToGetRequestInformation(ctx context.Context, requestConfiguration *ProviderItemRequestBuilderGetRequestConfiguration)(*i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.RequestInformation, error) {
    requestInfo := i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.NewRequestInformationWithMethodAndUrlTemplateAndPathParameters(i2ae4187f7daee263371cb1c977df639813ab50ffa529013b7437480d1ec0158f.GET, m.BaseRequestBuilder.UrlTemplate, m.BaseRequestBuilder.PathParameters)
    if requestConfiguration != nil {
        requestInfo.Headers.AddAll(requestConfiguration.Headers)
        requestInfo.AddRequestOptions(requestConfiguration.Options)
    }
    requestInfo.Headers.TryAdd("Accept", "application/json")
    return requestInfo, nil
}
// WithUrl returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
// returns a *ProviderItemRequestBuilder when successful
func (m *ProviderItemRequestBuilder) WithUrl(rawUrl string)(*ProviderItemRequestBuilder) {
    return NewProviderItemRequestBuilder(rawUrl, m.BaseRequestBuilder.RequestAdapter);
}

File: javaclient/javaclient/item/ProviderItemRequestBuilder.java
package javaclient.item;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.BaseRequestConfiguration;
import com.microsoft.kiota.HttpMethod;
import com.microsoft.kiota.RequestAdapter;
import com.microsoft.kiota.RequestInformation;
import com.microsoft.kiota.RequestOption;
import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParsableFactory;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
import javaclient.item.servicesjson.ServicesJsonRequestBuilder;
import javaclient.models.APIs;
/**
 * Builds and executes requests for operations under /{provider-id}
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ProviderItemRequestBuilder extends BaseRequestBuilder {
    /**
     * The servicesJson property
     * @return a {@link ServicesJsonRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ServicesJsonRequestBuilder servicesJson() {
        return new ServicesJsonRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link ProviderItemRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ProviderItemRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/{provider%2Did}", pathParameters);
    }
    /**
     * Instantiates a new {@link ProviderItemRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ProviderItemRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/{provider%2Did}", rawUrl);
    }
    /**
     * List all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
     * @return a {@link APIs}
     */
    @jakarta.annotation.Nullable
    public APIs get() {
        return get(null);
    }
    /**
     * List all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link APIs}
     */
    @jakarta.annotation.Nullable
    public APIs get(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = toGetRequestInformation(requestConfiguration);
        return this.requestAdapter.send(requestInfo, null, APIs::createFromDiscriminatorValue);
    }
    /**
     * List all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation() {
        return toGetRequestInformation(null);
    }
    /**
     * List all APIs in the directory for a particular providerNameReturns links to the individual API entry for each API.
     * @param requestConfiguration Configuration for the request such as headers, query parameters, and middleware options.
     * @return a {@link RequestInformation}
     */
    @jakarta.annotation.Nonnull
    public RequestInformation toGetRequestInformation(@jakarta.annotation.Nullable final java.util.function.Consumer<GetRequestConfiguration> requestConfiguration) {
        final RequestInformation requestInfo = new RequestInformation(HttpMethod.GET, urlTemplate, pathParameters);
        requestInfo.configure(requestConfiguration, GetRequestConfiguration::new);
        requestInfo.headers.tryAdd("Accept", "application/json");
        return requestInfo;
    }
    /**
     * Returns a request builder with the provided arbitrary URL. Using this method means any other path or query parameters are ignored.
     * @param rawUrl The raw URL to use for the request builder.
     * @return a {@link ProviderItemRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ProviderItemRequestBuilder withUrl(@jakarta.annotation.Nonnull final String rawUrl) {
        Objects.requireNonNull(rawUrl);
        return new ProviderItemRequestBuilder(rawUrl, requestAdapter);
    }
    /**
     * Configuration for the request such as headers, query parameters, and middleware options.
     */
    @jakarta.annotation.Generated("com.microsoft.kiota")
    public class GetRequestConfiguration extends BaseRequestConfiguration {
    }
}
